### ISSUE TITLE
It would be nice to have some bash completion for renku commands

### ISSUE THREAD

**Issue by jachro** [SwissDataScienceCenter/renku-python#1084]
As a Renku CLI user, I'd like to have some bash-completion for Renku commands so when I press the <TAB> key I either get the command I started typing completed or when I pressed <TAB> twice I get shown the possibilities I can choose from.

Acceptance criteria:
* when there's a single possibility to finish the command I've typed in and I pressed the <TAB> key, the command should be completed.
Example: 
```
%> renku datas + <TAB>
%> renku dataset 
```
* when there are multiple possibilities to what I've typed in and I pressed <TAB> twice, I should see the available options.
Example:
```
%> renku dataset + <TAB> + <TAB>
add create
%> renku dataset 
```
* the completion should work for project specific keywords/values like dataset names.

**Comment by rokroskar** [comment]
This was done in https://github.com/SwissDataScienceCenter/renku-python/pull/351 

After running 

```
renku --install-completion
```

you get

```
$ renku <tab> <tab>
clone     dataset   githooks  init      migrate   rerun     run       status    update
config    doctor    help      log       mv        rm        show      storage   workflow
$ renku dataset <tab> <tab>
add       create    edit      export    import    ls-files  ls-tags   rm        rm-tags   tag       unlink    update
```

However, afaik it does not autocomplete e.g. dataset names. And, as always, this feature could benefit from better documentation!

**Comment by jachro** [comment]
Oh cool! Could we add it to the templates by default maybe?

**Comment by rokroskar** [comment]
That's a good idea! I haven't tried using the auto-complete in a long time so I'm a bit worried unleashing it on the users unvetted, but maybe "it just works"?

**Comment by jachro** [comment]
It looks it works although it's somehow slow. And it would be nice if it's able to complete the project-specific keywords/values as I mentioned.

**Comment by rokroskar** [comment]
I agree - the slowness comes from the general slowness of the CLI... it also depends how you install it, if it's installed from pip directly it's a bit faster.

**Comment by rokroskar** [comment]
I am moving this issue to the [renkulab-docker](https://github.com/swissdatasciencecenter/renkulab-docker) repo - we should add the auto-completion to the entrypoint and/or `Dockerfile`.
